config/initializers/mas_cms_client.rb

```diff
@@ -1,8 +1,11 @@
 Mas::Cms::Client.config do |c|
-  c.timeout =  ENV['HTTP_REQUEST_TIMEOUT'].to_i
-  c.open_timeout =  ENV['HTTP_REQUEST_TIMEOUT'].to_i
+  c.timeout = ENV['HTTP_REQUEST_TIMEOUT'].to_i
+  c.open_timeout = ENV['HTTP_REQUEST_TIMEOUT'].to_i
   c.host =  ENV['FINCAP_CMS_URL']
   c.retries = 1
   c.cache = Rails.cache
-  c.cache_gets  = false
+  c.cache_gets = false
+end
+
+class Mas::Cms::Insight < Mas::Cms::Document
```

Comment:
I added the doubt section on the PR description but I will add here as well.

I don't know the best way to handle the class Mas::Cms::insight.
**I know that I could move to the gem and everything will be perfect** but what I don't like is:

1) Every time we need another page type we need to create a simple class on the gem.
2) The class does not do anything. Acctually it only exists to call /:locale/**insights**/:slug
3) On evaluation we need to do the same, on review we need to do the same and who knows the other templates that we will have.

Do someone has a better idea?

Reply:
if we didn't have this, and just called `document = Mas::Cms::Document.find(params[:id])` within the insights controller, would it not work?

Reply:
No will not work. It will request to /:locale/**documents**/:slug.